src/psyclone/configuration.py

```diff
@@ -748,27 +748,76 @@ class DynConfig(APISpecificConfig):
     # pylint: disable=too-few-public-methods
     def __init__(self, config, section):
         super(DynConfig, self).__init__(section)
-        self._config = config  # Ref. to parent Config object
-        try:
-            self._compute_annexed_dofs = section.getboolean(
-                'COMPUTE_ANNEXED_DOFS')
-        except ValueError as err:
-            raise ConfigurationError(
-                "error while parsing COMPUTE_ANNEXED_DOFS in the [dynamo0.3] "
-                "section of the config file: {0}".format(str(err)),
-                config=self._config)
+        # Ref. to parent Config object
+        self._config = config
+        # Initialise redundant computation setting
+        self._compute_annexed_dofs = None
+        # Initialise LFRic datatypes' default precisions settings
+        self._default_precision = {}
+
+        for key in section.keys():
+            # Do not handle any keys from the DEFAULT section
+            # since they are handled by Config(), not this class.
+            if key in config.get_default_keys():
+                continue
+            if key == "access_mapping":
+                # Handled in the base class APISpecificConfig
+                pass
+            elif key == "compute_annexed_dofs":
+                try:
+                    # Parse setting for redundant computation over annexed dofs
+                    self._compute_annexed_dofs = section.getboolean(
+                        'COMPUTE_ANNEXED_DOFS')
+                except ValueError as err:
+                    raise ConfigurationError(
+                        "error while parsing COMPUTE_ANNEXED_DOFS in the "
+                        "[dynamo0.3] section of the config file: {0}".
+                        format(str(err)), config=self._config)
+            elif key == "default_precision":
+                # Set default precisions from config file
+                all_prec = self.create_dict_from_string(section[key])
+                from psyclone.dynamo0p3 import SUPPORTED_FORTRAN_DATATYPES as \
+                    dynamo0p3_datatypes
+                for dtype in all_prec:
+                    if dtype not in dynamo0p3_datatypes:
+                        raise ConfigurationError(
+                            "Invalid datatype \"{0}\" found in the "
+                            "\"[dynamo0.3]\" section of the configuration "
+                            "file \"{1}\".".format(dtype, config.filename))
+                    if not all_prec[dtype]:
```

Comment:
It could happen by error when editing config file. I did comparison of all in the end as you suggested, but I used a list rather than a set as the list preserves order. I did have to use "filter" function on the lists to remove empty string values generated for `None`.